src/Interpreters/InterpreterExplainQuery.cpp

```diff
@@ -313,11 +333,33 @@ BlockInputStreamPtr InterpreterExplainQuery::executeImpl()
             plan.explainPipeline(buf, settings.query_pipeline_options);
         }
     }
+    else if (ast.getKind() == ASTExplainQuery::QueryEstimates)
+    {
+        if (!dynamic_cast<const ASTSelectWithUnionQuery *>(ast.getExplainedQuery().get()))
+            throw Exception("Only SELECT is supported for EXPLAIN ESTIMATES query", ErrorCodes::INCORRECT_QUERY);
 
-    if (single_line)
-        res_columns[0]->insertData(buf.str().data(), buf.str().size());
-    else
-        fillColumn(*res_columns[0], buf.str());
+        auto settings = checkAndGetSettings<QueryPlanSettings>(ast.getSettings());
+        QueryPlan plan;
+
+        // It should output the result even the format is Null, for example EXPLAIN ESTIMATES select * from x format Null;
+        if (ast.format && ast.format->getColumnName() == "Null")
```

Comment:
I found it controversial.
User may want to output result of EXPLAIN in any format and I would like to not make an exception for format Null.